p2p/server.go

```diff
@@ -591,11 +594,12 @@ func (srv *Server) setupDiscovery() error {
 			sconn = &sharedUDPConn{conn, unhandled}
 		}
 		cfg := discover.Config{
-			PrivateKey:  srv.PrivateKey,
-			NetRestrict: srv.NetRestrict,
-			Bootnodes:   srv.BootstrapNodes,
-			Unhandled:   unhandled,
-			Log:         srv.log,
+			PrivateKey:       srv.PrivateKey,
+			NetRestrict:      srv.NetRestrict,
+			Bootnodes:        srv.BootstrapNodes,
+			Unhandled:        unhandled,
+			Log:              srv.log,
+			ValidatedAddress: srv.Config.ValidatedAddress,
```

Comment:
So you have to config ValidatedAddress in config file? 
I guess we can read it from smartcontract for valid bird?

Reply:
The validated address will be read by the dove bird, and passed to here when boot up. I could switch to channel when we starting to support listening, and adding validated bird dynamically.